File: Array/Python/5-Two_Sum.py
# Two Sum
# Question Link - https://leetcode.com/problems/two-sum/description/
# Time Complexity - O(n)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        mpp = {}

        for i in range(len(nums)):
            ans = target - nums[i]
            if ans in mpp:
                return [mpp[ans], i]
            else:
                mpp[nums[i]] = i

        return []

File: Array/Cpp/5-Two_Sum.cpp
// Two Sum
// Question Link - https://leetcode.com/problems/two-sum/description/
// Time Complexity - O(n)

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> mpp;

        for(int i=0; i<nums.size(); i++){

            int ans = target - nums[i];

            if(mpp.find(ans) != mpp.end()){
                return {mpp[ans], i};
            }
            else{
                mpp[nums[i]] = i;
            }
        }

        return {};
    }
};
